src/react-components/ui-root.js

```diff
@@ -1043,44 +1043,62 @@ class UIRoot extends Component {
           />
         </div>
 
-        {!this.state.waitingOnAudio ? (
-          <div className={entryStyles.buttonContainer}>
-            {!isMobileVR && (
+        {!this.state.waitingOnAudio &&
+          !this.props.entryDisallowed && (
+            <div className={entryStyles.buttonContainer}>
+              {!isMobileVR && (
+                <a
+                  onClick={e => {
+                    e.preventDefault();
+                    this.attemptLink();
+                  }}
+                  className={classNames([entryStyles.secondaryActionButton, entryStyles.wideButton])}
+                >
+                  <FormattedMessage id="entry.device-medium" />
+                  <div className={entryStyles.buttonSubtitle}>
+                    <FormattedMessage
+                      id={isMobile ? "entry.device-subtitle-mobile" : "entry.device-subtitle-desktop"}
+                    />
+                  </div>
+                </a>
+              )}
               <a
                 onClick={e => {
                   e.preventDefault();
-                  this.attemptLink();
+
+                  if (promptForNameAndAvatarBeforeEntry || !this.props.forcedVREntryType) {
+                    this.props.hubChannel.sendEnteringEvent();
+
+                    const stateValue = promptForNameAndAvatarBeforeEntry ? "profile" : "device";
+                    this.pushHistoryState("entry_step", stateValue);
+                  } else {
+                    this.handleForceEntry();
+                  }
                 }}
-                className={classNames([entryStyles.secondaryActionButton, entryStyles.wideButton])}
-              >
-                <FormattedMessage id="entry.device-medium" />
-                <div className={entryStyles.buttonSubtitle}>
-                  <FormattedMessage id={isMobile ? "entry.device-subtitle-mobile" : "entry.device-subtitle-desktop"} />
-                </div>
-              </a>
-            )}
-            {promptForNameAndAvatarBeforeEntry || !this.props.forcedVREntryType ? (
-              <StateLink
-                stateKey="entry_step"
-                stateValue={promptForNameAndAvatarBeforeEntry ? "profile" : "device"}
-                history={this.props.history}
                 className={classNames([entryStyles.actionButton, entryStyles.wideButton])}
               >
                 <FormattedMessage id="entry.enter-room" />
-              </StateLink>
-            ) : (
+              </a>
+            </div>
+          )}
+        {this.props.entryDisallowed &&
+          !this.state.waitingOnAudio && (
+            <div className={entryStyles.buttonContainer}>
               <a
                 onClick={e => {
                   e.preventDefault();
-                  this.handleForceEntry();
+                  this.setState({ watching: true });
```

Comment:
Was this `watching` state meant to be used somewhere?

Reply:
yeah this is used to push you into the observer lobby mode, which is also triggered by the X in the panel

Reply:
Ah, okay, I just thought it was a new state variable 👍